tls/s2n_tls13_secrets.c

```diff
@@ -100,6 +100,28 @@ S2N_RESULT s2n_tls13_empty_transcripts_init()
     return S2N_RESULT_OK;
 }
 
+static S2N_RESULT s2n_calculate_transcript_digest(struct s2n_connection *conn)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(conn->handshake.hashes);
+
+    s2n_hash_algorithm hash_algorithm = S2N_HASH_NONE;
+    RESULT_ENSURE_REF(conn->secure.cipher_suite);
+    RESULT_GUARD_POSIX(s2n_hmac_hash_alg(conn->secure.cipher_suite->prf_alg, &hash_algorithm));
+
+    uint8_t digest_size = 0;
+    RESULT_GUARD_POSIX(s2n_hash_digest_size(hash_algorithm, &digest_size));
+
+    struct s2n_blob digest = { 0 };
+    RESULT_GUARD_POSIX(s2n_blob_init(&digest, CONN_HASHES(conn)->transcript_hash_digest, digest_size));
+
+    struct s2n_hash_state *hash_state = &conn->handshake.hashes->hash_workspace;
+    RESULT_GUARD(s2n_handshake_copy_hash_state(conn, hash_algorithm, hash_state));
+    RESULT_GUARD_POSIX(s2n_hash_digest(hash_state, digest.data, digest.size));
```

Comment:
No, because it just uses the existing memory from the connection. Technically it does use heap memory, but freed during the lifecycle of the entire hash structure and definitely not owned here.